modules/jaeger-config-storage.adoc

```diff
@@ -157,6 +118,152 @@ Minimum deployment = 16Gi*
+|`true`/ `false`
+|`true`
+
+|esIndexCleaner:
+  numberOfDays:
+|Number of days to wait before deleting an index.
+|Integer value
+|`7`
+
+|esIndexCleaner:
+  schedule:
+|Defines the schedule for how often to clean the Elasticsearch index.
+|Cron expression
+|"55 23 * * *"
+
+|esRollover:
+  schedule:
+|Defines the schedule for how often to rollover to a new Elasticsearch index.
+|Cron expression
+|'*/30 * * * *'
+|===
+
+.Production storage example
+[source,yaml]
+----
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simple-prod
+spec:
+  strategy: production
+  storage:
+    type: elasticsearch
+    elasticsearch:
+      nodeCount: 3
+      resources:
+        requests:
+          cpu: 1
+          memory: 16Gi
+        limits:
+          memory: 16Gi
+----
+
+.Storage example with persistent storage:
+[source,yaml]
+----
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simple-prod
+spec:
+  strategy: production
+  storage:
+    type: elasticsearch
+    elasticsearch:
+      nodeCount: 1
+      storage: # <1>
+        storageClassName: gp2
+        size: 5Gi
+      resources:
+        requests:
+          cpu: 200m
+          memory: 4Gi
+        limits:
+          memory: 4Gi
+      redundancyPolicy: ZeroRedundancy
+----
+
+<1> Persistent storage configuration. In this case AWS `gp2` with `5Gi` size. When no value is specified, Jaeger uses `emptyDir`. The Elasticsearch Operator provisions `PersistentVolumeClaim` and `PersistentVolume` which are not removed with Jaeger instance. You can mount the same volumes if you create a Jaeger instance with the same name and namespace.
+
+
+[id="jaeger-config-external-es_{context}"]
+== Connecting to an existing Elasticsearch instance
+
+Jaeger also allows you to use an existing (self-provisioned) Elasticsearch cluster for storage.  You do this by specifying the URL of the existing cluster as the`spec:storage:es:server-urls` value in your configuration.
+
+.Restrictions
+
+* There can be only one Jaeger with self-provisioned Elasticsearch instance per namespace.
+
+[NOTE]
+====
+Red Hat does not provide support for your external Elasticsearch instance. You can review the tested integrations matrix on the link:https://access.redhat.com/articles/5381021[Customer Portal].
+====
+
+The following configuration parameters are for an _external_ Elasticsearch instance, that is, an instance that was not created using the Elasticsearch Operator. You specify configuration options for external Elasticsearch under `spec:storage:options:es` in your configuration file.
+
+.ES configuration parameters
+[options="header"]
+[cols="l, a, a, a"]
+|===
+|Parameter |Description |Values |Default value
+4+|General Elasticsearch configuration settings
+|es:
+  server-urls:
+|URL of the Elasticsearch instance.
+|The fully-qualified domain name of the Elasticsearch server.
+|`http://elasticsearch.<namespace>.svc:9200`
+|es:
+  max-num-spans:
```

Comment:
Note, in the upstream docs "(deprecated, will be removed in release v1.21.0. Please use es.max-doc-count). The maximum number of spans to fetch at a time per query in Elasticsearch. The lesser of es.max-num-spans and es.max-doc-count will be used if both are set."

Reply:
So should we leave max-num-spans in with the deprecation notice,?  Or take it out and only document es.max-doc-count?

Reply:
@jpkrohling thoughts?
I think probably better to include the deprecation notice, as although the operator will handle the upgrade case, users may still try to use an old CR and deploy with 1.20GA.